app/add/page.tsx

```diff
@@ -1,168 +1,168 @@
-"use client";
+'use client';
 
 import {
-	Box,
-	Button,
-	Card,
-	Container,
-	FormControl,
-	FormLabel,
-	InputLabel,
-	MenuItem,
-	Select,
-	Stack,
-	TextField,
-	Typography,
-} from "@mui/material";
-import { useTaskContext } from "../context/TaskContext";
-import { useState } from "react";
-import { useRouter } from "next/navigation";
+  Box,
+  Button,
+  Card,
+  Container,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  Stack,
+  TextField,
+  Typography
+} from '@mui/material';
+import { useTaskContext } from '../context/TaskContext';
+import { useForm } from 'react-hook-form';
+import { useRouter } from 'next/navigation';
 
-export default function Add() {
-	const { addTask } = useTaskContext();
-
-	const [type, setType] = useState("");
-	const [status, setStatus] = useState("");
-	const [errors, setErrors] = useState<{ [key: string]: string }>({});
-
-	const router = useRouter();
+interface FormValues {
+  title: string;
+  description: string;
+  type: string;
+  status: string;
+}
 
-	const handleValidation = (title: string, description: string) => {
-		const newErr: { [key: string]: string } = {};
-		if (title === "") {
-			newErr.title = "Title is required.";
-		} else if (title.length > 20) {
-			newErr.title = "Title can't be more then 20 characters long.";
-		}
+export default function Add() {
+  const { addTask } = useTaskContext();
+  const router = useRouter();
 
-		if (description.length > 200) {
-			newErr.description =
-				"Description can't be more then 200 characters long.";
-		}
-		return newErr;
-	};
+  // Set up react-hook-form
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<FormValues>();
 
-	const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
-		event.preventDefault();
+  const onSubmit = (data: FormValues) => {
+    addTask({
+      id: Math.random().toString(36).slice(2, 9),
+      title: data.title,
+      description: data.description,
+      type: data.type as 'feature' | 'bug' | 'enhancement' | 'task',
+      status: data.status as 'open' | 'in-progress' | 'closed'
+    });
+    router.push('/');
```

Comment:
why?

Reply:
export interface Task {
  id: string;
  title: string;
  description: string | null;
  status: 'open' | 'in-progress' | 'closed' | null;
  type: 'feature' | 'bug' | 'enhancement' | 'task' | null;
}

This is my Task type therefore I need to cast a string to the  'open' | 'in-progress'... type. Otherwise eslint will be throwing an error.

Or are you asking about redirect? I redirect so that after you add a task, you will be redirected to see the whole task list